HelpSource/getSpec.schelp

```diff
@@ -0,0 +1,98 @@
+title:: getSpec
+
+summary:: Using Specs for proxy parameters: getSpec, addSpec, and specs
+categories:: Guides
+related:: Classes/NodeProxy, Classes/PatternProxy, Classes/Halo, Classes/Object
+
+section:: Using Specs for proxy parameters: getSpec, addSpec, and specs
+
+JITLib proxies like PatternProxy (and thus TaskProxy, Pdef, and Tdef)
+and NodeProxy/Ndef and other objects have numerical settings, for which
+defining a Spec is useful: a practical mininum, maximum and curve (warp).
+
+getSpec and addSpec provide a general interface for proxies and
+all other objects that use .set methods for named parameters.
+This makes it easy to set and access ranges for parameters,
+e.g. for GUIs and other interfaces:
+
+code::
+Tdef(\x).set(\myfreq, 345);
+Tdef(\x).addSpec(\myfreq, [30, 3000, \exp]);
+Tdef(\x).gui; // gui uses myfreq spec automatically
+// specs are accessed with:
+Tdef(\x).getSpec(\myfreq);
+// or this gets a dict with all known specs
+Tdef(\x).getSpec;
+// when you change a spec later to tune it,
+// the gui immediately uses it.
+Tdef(\x).addSpec(\myfreq, [50, 5000, \exp]);
+// specs are also used for setting params with unipolar values:
+Tdef(\x).setUni(\myfreq, 0.5); // center of range is 500
+Tdef(\x).getUni(\myfreq);
+// this guarantees that values stay within the range,
+// and is easy to use with controllers, e.g. hardware sliders.
+::
+
+For historical reasons, NodeProxy (and Ndef) has two ways to define a spec for a given parameter name:
+
+code::
+// by implicit spec in the synth function(s), as in
+Ndef(\x, { Pulse.ar(\freqx.kr(200, spec: [3, 300, \exp])) })
+
+// and in the general JITLib way by addSpec (Halo).
+Ndef(\x).addSpec(\freqx, [3, 300, \exp])
+::
+
```

Comment:
Maybe it is good to mention that `addSpec` is a feature of `JITLibExtensions` (it is not there without it),

Reply:
done, explained above too